### PR TITLE
ebpf: memorize what syscalls were notified

### PR DESCRIPTION
**Pull request by giuseppe**
avoid generating too many perf events by memorizing what syscalls were
already seen and notified.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
LGTM, great idea!

**Comment by vrothberg** [comment]
@rhatdan PTAL

File: ebpf.go [hunk 1]
```diff
@@ -64,13 +66,21 @@ int enter_trace(struct tracepoint__raw_syscalls__sys_enter* args)
...
     struct syscall_data data = {};
     u64 key = 0;
+    u32 seen = 0xFF;
     unsigned int zero = 0;
     struct task_struct *task;
     struct nsproxy *nsproxy;
     struct mnt_namespace *mnt_ns;
+    int id = (int)args->id;
+
+    if (seen_syscalls.lookup(&id) == seen) {
```
Comment (rhatdan) [inline ebpf.go:76]: lookup returne 0xFF on success?

File: ebpf.go [hunk 2]
```diff
@@ -64,13 +66,21 @@ int enter_trace(struct tracepoint__raw_syscalls__sys_enter* args)
...
+	/* The syscall was already notified.  */
```
Comment (rhatdan) [inline ebpf.go:77]: Nit; Shouldn't this be // The syscall was already notified.

File: ebpf.go (hunk 1 above)
Comment (rhatdan) [reply ebpf.go:76]: Ok any reason to select 0xFF rather then just 1?

**Comment by rhatdan** [comment]
LGTM

File: ebpf.go (hunk 1 above)
Comment (giuseppe) [reply ebpf.go]: I've simplified it to not use `0xFF` but the syscall number value itself, as anyway the value is ignored

File: ebpf.go (hunk 2 above)
Comment (giuseppe) [reply ebpf.go]: fixed

**Comment by giuseppe** [comment]
tests are green

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**